components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ThrottleHandler.java

```diff
@@ -833,6 +861,83 @@ private OMElement createSpikeArrestSubscriptionLevelPolicy(String policyName, in
         return parsedPolicy;
     }
 
+    private OMElement createSpikeArrestApplicationLevelPolicy(String policyName, int maxCount, int unitTime) {
+
+        OMElement parsedPolicy = null;
+        StringBuilder policy = new StringBuilder(APIThrottleConstants.WS_THROTTLE_POLICY_HEADER);
+        if (maxCount != 0 && unitTime != 0) {
+            policy.append(createPolicyForRole(policyName, Integer.toString(unitTime), Integer.toString(maxCount)));
+        }
+        policy.append(APIThrottleConstants.WS_THROTTLE_POLICY_BOTTOM);
+        try {
+            parsedPolicy = AXIOMUtil.stringToOM(policy.toString());
+        } catch (XMLStreamException e) {
+            log.error("Error occurred while creating policy file for Hard Throttling.", e);
```

Comment:
add a reason why this exception is ignored (without throwing an exception and stopping the flow)

Reply:
added in b56bf3020abe507022f90c1b46e64547b10cf7ac